src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -702,6 +716,21 @@ public long ramBytesUsed() {
         public String toString() {
             return "[id: " + translogId + ", location: " + translogLocation + ", size: " + size + "]";
         }
+
+        @Override
+        public int compareTo(Location o) {
+            if (translogId == o.translogId) {
+                return Long.compare(translogLocation, o.translogLocation);
+            }
+            return Long.compare(translogId, o.translogId);
+        }
+
```

Comment:
If we are going to add Comparable, can we also define hashcode/equals consistently with it?